src/actions/EditNew.ts

```diff
@@ -38,49 +37,23 @@ class EditNew implements Action {
 
     const editor = ensureSingleEditor(targets);
 
-    const richTargets: RichTarget[] = targets.map((target) => {
-      const context = target.getEditNewContext(this.isBefore);
-      const common = {
-        target,
-        cursorRange: target.contentRange,
-      };
-      switch (context.type) {
-        case "command":
-          return {
-            ...common,
-            type: "command",
-            command: context.command,
-          };
-        case "delimiter":
-          return {
-            ...common,
-            type: "delimiter",
-            delimiter: context.delimiter,
-          };
-      }
-    });
+    let state: State = {
+      targets,
+      thatRanges: targets.map(({ thatTarget }) => thatTarget.contentRange),
+      // The range updater doesn't work with sparsely populated arrays. All these values will get over written.
+      cursorRanges: new Array(targets.length).fill(new Range(0, 0, 0, 0)),
```

Comment:
as mentioned below, I'd fill these with `undefined` and then filter them out / track indices in `runDelimiterTargets` before updating and returning state